pkgs/os-specific/linux/lvm2/default.nix

```diff
@@ -67,23 +67,14 @@ stdenv.mkDerivation rec {
     sed -i 's|^#define LVM_CONFIGURE_LINE.*$|#define LVM_CONFIGURE_LINE "<removed>"|g' ./include/configure.h
   '';
 
-
-  patches = lib.optionals stdenv.hostPlatform.isMusl [
-    (fetchpatch {
-      name = "fix-stdio-usage.patch";
-      url = "https://git.alpinelinux.org/aports/plain/main/lvm2/fix-stdio-usage.patch?h=3.7-stable&id=31bd4a8c2dc00ae79a821f6fe0ad2f23e1534f50";
-      sha256 = "0m6wr6qrvxqi2d2h054cnv974jq1v65lqxy05g1znz946ga73k3p";
-    })
+  patches = [
+    # Musl fixes from Alpine.
+    ./fix-stdio-usage.patch
```

Comment:
Why apply this unconditionally? Without this, wouldn't this be a noop for the non-musl case?

Reply:
Conditionally applied patches mean increased maintenance burden.  It's the reason these patches were able to bitrot in the first place.  This PR is to staging, where there's almost always a full stdenv rebuild anyway, so it doesn't matter at all whether it's a no-op.

Reply:
Alright :+1:

I also meant to ask/say that if this was conditional, we could just go directly to master with it.

I've been trying to always test this when updating/modifying lvm2, but iirc it was mostly broken on master already and I don't use musl myself, so I didn't fix it. Sorry if I broke this.

Anyways, lets go ahead with this then. Is there a musl team or can I ping you if I run into problems with this in the future?

Reply:
There's an exotic platform team that I'm in, maybe that's the one to use?